site/docs/starter/slash-commands.md

```diff
@@ -0,0 +1,550 @@
+    // We should check if the command defines a specific scope, if not check the `else`
+    if (command.scope) {
+      // Push guild commands
+      if (command.scope === "Guild") {
+        perGuildCommands.push({
+          name: command.name,
+          description: command.description,
+          type: command.type,
+          options: command.options ? command.options : undefined,
+        });
+        // Push Global (all guilds + DMs) commands
+      } else if (command.scope === "Global") {
+        globalCommands.push({
+          name: command.name,
+          description: command.description,
+          type: command.type,
+          options: command.options ? command.options : undefined,
+        });
+      }
+    } else {
+      // Push commands without a scope to every guild (all guilds, no DMs)
+      perGuildCommands.push({
+        name: command.name,
+        description: command.description,
+        type: command.type,
+        options: command.options ? command.options : undefined,
+      });
+    }
+  }
+
+  if (globalCommands.length && (scope === "Global" || scope === undefined)) {
+    // Due to the way discord internally handles global commands, it might take up to an hour before they are available.
+    log.info(
+      "Updating Global Commands, changes should apply shortly...",
+    );
+    // Upsert will add commands, replace existing commands, and delete ones that do not exist anymore.
+    await bot.helpers.upsertApplicationCommands(globalCommands).catch(
+      log.error,
+    );
+  }
+
+  if (perGuildCommands.length && (scope === "Guild" || scope === undefined)) {
+    await bot.guilds.forEach(async (guild: Guild) => {
+      // Upsert will add commands, replace existing commands, and delete ones that do not exist anymore.
+      await upsertApplicationCommands(bot, perGuildCommands, guild.id);
+    });
+  }
+}
+
+/** Update commands for a guild */
+export async function updateGuildCommands(bot: Bot, guild: Guild) {
+  const perGuildCommands: MakeRequired<CreateApplicationCommand, "name">[] = [];
+
+  // Goes over all commands in the command map, so it's important to load the commands before running this function.
+  for (const command of commands.values()) {
+    if (command.scope) {
+      // Since we are only updating guild commands, we should only select commands with the "Guild" scope.
+      if (command.scope === "Guild") {
+        perGuildCommands.push({
+          name: command.name,
+          description: command.description,
+          type: command.type,
+          options: command.options ? command.options : undefined,
+        });
+      }
+    }
+  }
+
+  if (perGuildCommands.length) {
+    // Upsert will add commands, replace existing commands, and delete ones that do not exist anymore.
+    await upsertApplicationCommands(bot, perGuildCommands, guild.id);
+  }
+}
+// ...
+```
+
+We'll add this to our `mod.ts` to make sure that we always have the latest commands ready for users:
```

Comment:
I would recommend rephrasing this so that its done on mod.ts so that when u make any changes it will automatically make changes once u restart ur bot.

Reply:
I'll update the guide to use `deno run --watch`